src/js/libcs/OpDrawing.js

```diff
@@ -281,53 +281,49 @@ eval_helper.drawcircle = function(args, modifs, df) {
     var v0 = evaluateAndVal(args[0]);
     var v1 = evaluateAndVal(args[1]);
 
-    function magic_circle(ctx, x, y, r) {
-        m = 0.551784;
-
-        ctx.save();
-        ctx.translate(x, y);
-        ctx.scale(r, r);
-
-        ctx.beginPath();
-        ctx.moveTo(1, 0);
-        ctx.bezierCurveTo(1, -m, m, -1, 0, -1);
-        ctx.bezierCurveTo(-m, -1, -1, -m, -1, 0);
-        ctx.bezierCurveTo(-1, m, -m, 1, 0, 1);
-        ctx.bezierCurveTo(m, 1, 1, m, 1, 0);
-        ctx.closePath();
-        ctx.restore();
-    }
-
-
     var pt = eval_helper.extractPoint(v0);
 
 
-    if (!pt.ok || v1.ctype !== 'number') {
+    if (!pt.ok || v1.ctype !== 'number' || !CSNumber._helper.isAlmostReal(v1) || !List._helper.isAlmostReal(v0)) {
```

Comment:
On 02/17/2016 10:41 AM, Martin von Gagern wrote:
>
> In src/js/libcs/OpDrawing.js
> <https://github.com/CindyJS/CindyJS/pull/259#discussion_r53140238>:
>
> >      var pt = eval_helper.extractPoint(v0);
> >  
> >  
> > -    if (!pt.ok || v1.ctype !== 'number') {
> > +    if (!pt.ok || v1.ctype !== 'number' || !CSNumber._helper.isAlmostReal(v1) || !List._helper.isAlmostReal(v0)) {
>
> I think the |List._helper.isAlmostReal(v0)| check is incorrect.
> Consider |drawcircle(i*[3,2,1], 4)| where the point is a complex
> representative of a real point. |extractPoint| should be able to deal
> with this, and report its success in the |pt.ok| flag. Checking
> whether the homogeneous coordinate vector is real as well causes
> unneccessary restrictions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CindyJS/CindyJS/pull/259/files#r53140238>.
>

Oh yes you're right!

-- 
Michael Strobel, M.Sc.

Technische Universität München -- Zentrum Mathematik
Lehrstuhl für Geometrie und Visualisierung (M10)
Boltzmannstr. 3, 85747 Garching bei München 
https://www-m10.ma.tum.de/bin/view/Lehrstuhl/MichaelStrobel